### PR TITLE
feature: add operation summary sentence casing rule

### PR DESCRIPTION
**Pull request by TaylorKrusen**
Added a configurable rule to check operation summaries for sentence casing

### CODE REVIEW

File: configurable-rules/operation-summary-sentence-case/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+**Why** -- "Sentence casing vs title casing" may be the "tabs vs spaces" of the writing world, but being _consistent_ is often more important than being right. At Redocly, we prefer sentence casing for Operation summaries. By adding this configurable rule to our linter, we ensure our entire team writes them that way.
+
+## Code
+
+Here's how the configurable rule looks in the `redocly.yaml` file:
+
+```yaml
+rules:  
+  rule/operation-summary-sentence-case:
+    subject: ``
```
Comment (adamaltman) [inline configurable-rules/operation-summary-sentence-case/README.md:19]:
```suggestion
    subject:
```

File: configurable-rules/operation-summary-sentence-case/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      type: Operation
+      property: summary
+    message: "Operation summary must be sentence cased."
+    assertions: 
+      pattern: /^[A-Z]+[^A-Z]+$/
```
Comment (adamaltman) [inline configurable-rules/operation-summary-sentence-case/README.md:24]: Should it be allowed to end with a period? or not? In our own rules, we disallow it.

Comment (TaylorKrusen) [reply configurable-rules/operation-summary-sentence-case/README.md:24]: Yes, I think so. That way we're only asserting one thing at a time.  I'd expect checking for ending with a period to be a separate rule. 
Maybe that's another good example we can add where users might chain configurable rules?

File: configurable-rules/operation-summary-sentence-case/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Operation summary uses sentence casing
+
+Authors:
+- [`@TaylorKrusen`](https://github.com/TaylorKrusen) Taylor Krusen (Redocly) 
+
+## What this does and why
+
+**What** -- this rule enforces a consistent patterns of capitalization on Operation summaries.
```
Comment (lornajane) [inline configurable-rules/operation-summary-sentence-case/README.md:8]:
```suggestion
**What** -- this rule enforces a consistent pattern of capitalization on Operation summaries.  
```

File: configurable-rules/operation-summary-sentence-case/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      property: summary
+    message: "Operation summary must be sentence cased."
+    assertions: 
+      pattern: /^[A-Z]+[^A-Z]+$/
+```
+
+> [!NOTE]
+This rule can be repurposed for other fields with a single sentence, but not multiple sentences.
+
+### Rule explanation
```
Comment (lornajane) [inline configurable-rules/operation-summary-sentence-case/README.md:30]: Blank line after heading please (it's a tiny thing, but some markdown tools will choke, and it's also easier to read the source with the extra vertical whitespace. We should probably have a linter for this!

**Review by lornajane — CHANGES_REQUESTED** [review]
I like this one! I added some nitpicks, and I'd also like to see something about what to do when there's a proper noun or some other legit reason for a capital letter in the summary. If the advice is to add an ignore, link to the docs on how to do that.

**Comment by lornajane** [comment]
You re-requested the review, but I would still like to see something in the page about what to do when there's a correct title that doesn't match the rule, so same feedback as before since I still don't see that here.

**Comment by TaylorKrusen** [comment]
Sorry @lornajane, I missed that note. Added an ignore section.

**Review by lornajane — APPROVED** [review]
Thanks for the addition, this is great!

**Pull request merged**